### PR TITLE
refactor: state signature collector proxy wiring

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
This PR creates proxy wiring for the `StateSignatureCollector`

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 1]
```diff
@@ -465,11 +467,42 @@ private void wire() {
...
+                    if (rs.get().isComplete()) {
+                        return true;
+                    } else {
+                        // close the second reservation on states that are not passed on.
+                        rs.close();
+                        return false;
+                    }
+                })
+                .buildAdvancedTransformer(new SignedStateReserver("completeStatesReserver"));
+        ;
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:503]: extra `;`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java [hunk 2]
```diff
@@ -465,11 +467,42 @@ private void wire() {
...
+
+        // Split output of StateSignatureCollector into single ReservedSignedStates.
+        OutputWire<ReservedSignedState> splitReservedSignedStateWire = stateSignatureCollectorWiring.getOutputWire().buildSplitter("reservedStateSplitter", "reserved state lists");
+        // Add another reservation to the signed states since we are soldering to two different input wires
+        OutputWire<ReservedSignedState> allReservedSignedStatesWire =
+                splitReservedSignedStateWire.buildAdvancedTransformer(new SignedStateReserver("allStatesReserver"));
+        // All reserved signed states are saved to disk.
+        allReservedSignedStatesWire.solderTo(signedStateFileManagerWiring.saveToDiskFilter());
+        // Filter to complete states only and add a 3rd reservation since completes states are used in two input wires.
+        OutputWire<ReservedSignedState> completeReservedSignedStatesWire = allReservedSignedStatesWire
```
Comment (cody-littley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:492]: Variables can be final

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java:503]: removed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformWiring.java]: made final

**Review by cody-littley — APPROVED** [review]

**Pull request merged**